.github/workflows/continuous-integration.yml

```diff
@@ -55,113 +55,114 @@ jobs:
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install poetry 🦄
+        uses: Gr1N/setup-poetry@v8
+
+      - name: Load Poetry Cached Libraries ⬇
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pypoetry
+          key: ${{ runner.os }}-poetry-${{ hashFiles('**/poetry.lock') }}
+          restore-keys: ${{ runner.os }}-poetry-
+
+      - name: Install dependencies 🖥
+        run: |
+          poetry install --no-interaction
+          poetry run pip install prompt_toolkit==${{ matrix.promttoolkit }}
+
+      - name: Test Code 🔍
+        run: make test
+
+      - name: Send Coverage Report 📊
+        shell: bash -l {0}
+        env:
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COVERALLS_SERVICE_NAME: github
+          COVERALLS_SERVICE_JOB_ID: ${{ github.run_id }}
+          COVERALLS_SERVICE_NUMBER: "${{ github.workflow }}-${{ github.run_number }}"
+          COVERALLS_PARALLEL: true
+        run: poetry run coveralls
   post-test:
     name: Signal test completion
     needs: test
     runs-on: ubuntu-latest
     container: python:3-slim
     steps:
-    - name: Finished
-      run: |
-        pip3 install --upgrade coveralls
-        coveralls --service=github --finish
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Finished
+        run: |
+          pip3 install --upgrade coveralls
+          coveralls --service=github --finish
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COVERALLS_SERVICE_NUMBER: "${{ github.workflow }}-${{ github.run_number }}"
 
   docs:
     name: Test Docs
     runs-on: ubuntu-latest
 
     steps:
-    - name: Checkout git repository 🕝
-      uses: actions/checkout@v3
+      - name: Checkout git repository 🕝
+        uses: actions/checkout@v3
 
-    - name: Set up Python 3.7 🐍
-      uses: actions/setup-python@v4
-      with:
-        python-version: 3.7
+      - name: Set up Python 3.11 🐍
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
 
-    - name: Install poetry 🦄
-      uses: Gr1N/setup-poetry@v7
+      - name: Install poetry 🦄
+        uses: Gr1N/setup-poetry@v8
 
-    - name: Load Poetry Cached Libraries ⬇
-      uses: actions/cache@v3
-      with:
-        path: ~/.cache/pypoetry
-        key: ${{ runner.os }}-poetry-${{ hashFiles('**/poetry.lock') }}
-        restore-keys: ${{ runner.os }}-poetry
+      - name: Load Poetry Cached Libraries ⬇
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pypoetry
+          key: ${{ runner.os }}-poetry-${{ hashFiles('**/poetry.lock') }}
+          restore-keys: ${{ runner.os }}-poetry
 
-    - name: Install dependencies 🖥
-      run: poetry install --no-interaction --extras "docs"
+      - name: Install dependencies 🖥
+        run: poetry install --no-interaction --with=docs
```

Comment:
Is this just a syntactic change?

Reply:
the spacing in this file is syntactic (I forgot to turn off my yaml linter).

Generally, `group` is used for internal stuff (like doc dependencies, dev dependencies, etc) with `with`, while `extras` is used for additional runtime features.